batchflow/models/torch/modules.py

```diff
@@ -226,3 +226,190 @@ def forward(self, x):
         return self.gamma*out + x
+
+
+class BAM(nn.Module):
+    """ Bottleneck Attention Module.
+
+    Jongchan Park. et al. "'BAM: Bottleneck Attention Module
+    <https://arxiv.org/abs/1807.06514>'_"
+
+    Parameters
+    ----------
+    ratio : int
+        Squeeze ratio for the number of filters.
+        Default is 16.
+    dilation_rate : int
+        The dilation rate in the convolutions in the spatial attention submodule.
+        Default is 4.
+    """
+    def __init__(self, inputs=None, ratio=16, dilation_rate=4, **kwargs):
+        super().__init__()
+        self.bam_attention = ConvBlock(
+            inputs=inputs, layout='S' + 'cna'*4 + '+ a',
+            filters=['same//{}'.format(ratio), 'same', 'same', 1], kernel_size=[1, 3, 3, 1],
+            dilation_rate=[1, dilation_rate, dilation_rate, 1], activation=['relu']*4+['sigmoid'],
+            squeeze_layout='Vfnaf', ratio=ratio, squeeze_activations='relu', **kwargs)
+        self.desc_kwargs = {
+            'class': self.__class__.__name__,
+            'in_filters': get_num_channels(inputs),
+            'out_filters': get_num_channels(inputs),
+            'dilation_rate': dilation_rate,
+            'ratio': ratio
+        }
+
+    def forward(self, x):
+        return x * (1 + self.bam_attention(x))
+
+    def __repr__(self):
+        layer_desc = ('{class}({in_filters}, {out_filters}, '
+                      'dilation_rate={dilation_rate}, ratio={ratio})').format(**self.desc_kwargs)
+        return layer_desc
+
+
+class CBAM(nn.Module):
+    """ Convolutional Block Attention Module.
+
+    Sanghyun Woo. et al. "'CBAM: Convolutional Block Attention Module
+    <https://arxiv.org/abs/1807.06521>'_"
+
+    Parameters
+    ----------
+    ratio : int
+        Squeeze ratio for the number of filters.
+        Default is 16.
+    pool_ops : list of str
+        Pooling operations for channel_attention module.
+        Default is `('avg', 'max')`.
+    """
+    def __init__(self, inputs=None, ratio=16, pool_ops=('avg', 'max'), **kwargs):
+        super().__init__()
+        self.channel_attention(inputs, ratio, pool_ops, **kwargs)
+        self.spatial_attention(inputs, **kwargs)
+        self.desc_kwargs = {
+            'class': self.__class__.__name__,
+            'in_filters': get_num_channels(inputs),
+            'out_filters': get_num_channels(inputs),
+            'pool_ops': ('avg', 'max'),
+            'ratio': ratio
+        }
+
+    def channel_attention(self, inputs, ratio, pool_ops, **kwargs):
+        " Channel attention module."
+        self.pool_layers = []
+        num_dims = get_num_dims(inputs)
+        num_channels = get_num_channels(inputs)
+
+        for pool_op in pool_ops:
+            pool = Pool(inputs=inputs, op=pool_op)
+            self.pool_layers.append(pool)
+
+        tensor = self.pool_layers[0](inputs)
+        self.share_layer = ConvBlock(inputs=tensor, layout='faf>',
```

Comment:
It is `shared_layer`, I think.
Naming can be improved but, anyways, this PR looks very clean and nice

Reply:
```suggestion
        self.shared_layer = ConvBlock(inputs=tensor, layout='faf>',
```